### PR TITLE
try length+data records, need an Iterator

### PR DESCRIPTION
**Pull request by rob-solana**

### CODE REVIEW

**Comment by rob-solana** [comment]
@garious I need help making this work.  I'm stuck on the fact that the iterator I'm returning is of an unknowable size.  I tried a Box(), but that didn't help.

**Comment by aeyakovenko** [comment]
When you process the ledger can you build a map of blob index -> byte offset

File: src/entry_writer.rs [hunk 1]
```diff
@@ -50,28 +57,50 @@ impl<'a, W: Write> EntryWriter<'a, W> {
...
+    reader: R,
+    entry_bytes: Vec<u8>,
+    len_len: u64,
 }
 
-/// Return an iterator for all the entries in the given file.
-pub fn read_entries<R: BufRead>(reader: R) -> impl Iterator<Item = io::Result<Entry>> {
-    reader.lines().map(|s| read_entry(&s?))
-}
+impl Iterator for EntryReader<BufRead> {
```
Comment (garious) [inline src/entry_writer.rs]: ```impl<R: BufRead> Iterator for EntryReader<R> {```

**Comment by aeyakovenko** [comment]
@rob-solana there is no final form for the code :).  so if we can unblock the networking problems, i don't see a reason not to.

**Comment by rob-solana** [comment]
moving to a binary ledger will allow testnet to run at least 4 or 5 times as long as today (6 hours) before running out of disk on a 2TB volume.

**Pull request merged**